packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -180,18 +233,45 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     });
   }
 
+  const getSelectedIndexes = () => {
+    const indexArray: number[] = [];
+    isMultipleEnabled(valueProp, multiple) &&
+      valueProp?.forEach((value) => {
+        indexArray.push(
+          optionsRef.current.findIndex((option) => option.value === value),
+        );
+      });
+    return indexArray;
+  };
+
   // Reducer where all the component-wide state is stored
   const [{ isOpen, selectedIndex, focusedIndex }, dispatch] = React.useReducer(
     comboBoxReducer,
     {
       isOpen: false,
-      selectedIndex: valueProp
+      selectedIndex: isMultipleEnabled(valueProp, multiple)
+        ? valueProp
+          ? getSelectedIndexes()
+          : []
+        : valueProp
```

Comment:
Would be good to explain/simplify what is going on here. I think we could avoid a nested ternary

Reply:
Second ternary is not needed because if `valueProp` is undefined, `getSelectedIndexes` will return empty array.

Reply:
Removed second ternary